src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx

```diff
@@ -58,12 +76,19 @@ export const SetPriorities: React.VFC<SetPrioritiesProps> = ({
     const availableInternalGroupPositions = internalGroupPositions.filter(
       position => !priorityIds.includes(position.id)
     )
-
+    handlers.setState(filteredPriorities)
     // Filtered positions are those which are available for the user to add to their priority listing
     setFilteredInternalGroupPositions(availableInternalGroupPositions)
-    setPriorities(filteredPriorities as InternalGroupPositionPriorityNode[])
-  }, [applicant.priorities, internalGroupPositions])
+  }, [internalGroupPositions, applicant.priorities])
 
+  useEffect(() => {
+    const filteredPriorities = values.filter(priority => priority !== null)
+    const priorityIds = filteredPriorities.map(
+      priority => priority!.internalGroupPosition.id
+    )
+    handleUpdatePriorities(priorityIds)
+    setIsDirty(false)
+  }, [isDirty])
```

Comment:
Does this need a useEffect? Internal state in most cases are manageable with eventhandlers

Reply:
I don't know. Couldn't make it work as I wanted it to when putting all the logic in a handler that both reorders and updates, possibly due to how `use-list-state`'s handler reorders elements.